### PR TITLE
stm32wlx i2c implementation

### PR DESCRIPTION
**Pull request by ofauchon**
This PR enables I2C buses on stm32wlx devices. 
It has been tested on GNSE devices and SHTC3 i2c sensor

### CODE REVIEW

File: src/machine/i2c.go [hunk 1]
```diff
@@ -1,5 +1,5 @@
-//go:build atmega || nrf || sam || (stm32 && !stm32wlx) || fe310 || k210 || rp2040
-// +build atmega nrf sam stm32,!stm32wlx fe310 k210 rp2040
+//go:build atmega || nrf || sam || fe310 || k210 || rp2040 || stm32wlx
```
Comment (kenbell) [inline src/machine/i2c.go]: This will disable for all other stm32?  Should just leave the stm32 clause?

**Review by kenbell — CHANGES_REQUESTED** [review]

File: src/machine/i2c.go (hunk 1 above)
Comment (ofauchon) [reply src/machine/i2c.go]: Was a mistake. fixed

**Comment by kenbell** [comment]
lgtm

**Review by kenbell — APPROVED** [review]

**Comment by deadprogram** [comment]
Thanks for implementing this @ofauchon and thanks for the review @kenbell 

Now merging.

**Pull request merged**